hips/draw/simple.py

```diff
@@ -1,5 +1,156 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """HiPS tile drawing -- simple method."""
+import os
+from pathlib import Path
+
+import healpy as hp
+import numpy as np
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+from skimage import transform as tf
+from typing import List
+
+from ..tiles import HipsSurveyProperties, HipsTile, HipsTileMeta
+from ..utils import WCSGeometry, compute_healpix_pixel_indices, boundaries
 
 __all__ = [
+    'draw_sky_image',
+    'make_sky_image',
 ]
+
+
+def draw_sky_image(geometry: WCSGeometry, tiles: List[HipsTile]) -> np.ndarray:
+    """Draw sky image using the simple and quick method.
+
+    Parameters
+    ----------
+    geometry : 'WCSGeometry'
+        An object of WCSGeometry
+    tiles : List[HipsTile]
+        A list of HipsTile
+
+    Returns
+    -------
+    np.ndarray
+        Returns a numpy array containing all HiPS tiles projected onto it
+    """
+    all_sky = np.zeros(geometry.shape)
+    for tile in tiles:
+        draw_tile = DrawSkyImageOneTile(geometry, tile, all_sky.shape)
+        draw_tile.compute_corners()
+        draw_tile.apply_projection()
+        all_sky += draw_tile.warp_image()
+    return all_sky
+
+
+class DrawSkyImageOneTile:
+    """A private function for drawing a tile over a sky image.
+
+    Parameters
+    ----------
+    geometry : 'WCSGeometry'
+        An object of WCSGeometry
+    tile : HipsTile
+       An object of HipsTile
+    shape ; tuple
+        Shape of the all-sky image
+    """
+
+    def __init__(self, geometry: WCSGeometry, tile: HipsTile, shape: tuple) -> None:
+        self.geometry = geometry
+        self.tile = tile
+        self.shape = shape
+        self.wcs = WCS(tile.header)
+        self.corners = None
+        self.pt = None
+
+    def compute_corners(self) -> None:
+        nside = hp.order2nside(self.tile.meta.order)
+        theta, phi = boundaries(nside, self.tile.meta.ipix)
+        radec = SkyCoord(ra=phi, dec=np.pi / 2 - theta, unit='radian', frame='icrs')
+        self.corners = []
+        for i in range(len(radec.ra.deg)):
+            self.corners.append([radec.ra.deg[i], radec.dec.deg[i]])
+
+    def apply_projection(self) -> None:
+        src = self.geometry.wcs.wcs_world2pix(self.corners, 0)
+        dst = np.array([[511, 0], [511, 511], [0, 511], [0, 0]])
```

Comment:
The number 511 shouldn't be hard-coded, but should be computed from the tile `shape`, no?
My understanding is that it is usually, but not always 512 x 512 pixels.
Maybe this `dst` array would make sense as a property on `HipsTileMeta`?